File: apps/qizhi/mobile/MentorAI-iOS/MentorAI/Chat/SessionAPI.swift
import Foundation

struct SessionAPI {
    let client: APIClient

    init(baseURL: URL) {
        self.client = URLSessionAPIClient(baseURL: baseURL)
    }

    init(client: APIClient) {
        self.client = client
    }

    func list(token: String) async throws -> [ChatSession] {
        try await client.get("/session/list", query: [], bearerToken: token)
    }

    func detail(id: String, token: String) async throws -> ChatSession {
        try await client.get("/session", query: [URLQueryItem(name: "id", value: id)], bearerToken: token)
    }

    func delete(id: String, token: String) async throws {
        try await client.delete("/session/delete", query: [URLQueryItem(name: "id", value: id)], bearerToken: token)
    }
}

File: apps/qizhi/mobile/MentorAI-Android/app/src/main/kotlin/com/mentorai/app/chat/SessionAPI.kt
package com.mentorai.app.chat

import com.mentorai.app.networking.APIClient

class SessionAPI(private val client: APIClient) {

    /** GET /session/list — summary list, no messages. */
    suspend fun list(token: String): List<ChatSession> =
        client.getEnvelope("/session/list", bearerToken = token)

    /** GET /session?id=... — full detail with messages. */
    suspend fun detail(id: String, token: String): ChatSession =
        client.getEnvelope("/session", mapOf("id" to id), token)

    /** DELETE /session/delete?id=... */
    suspend fun delete(id: String, token: String) {
        client.deleteEnvelope("/session/delete", mapOf("id" to id), token)
    }
}
